tests/common_testing.py

```diff
@@ -0,0 +1,35 @@
+from django.conf import settings
+from django.test import TestCase as DjangoTestCase
+from django.test import TransactionTestCase as DjangoTransactionTestCase
+
+
+class OAuth2ProviderTestCase(DjangoTestCase):
+    """Place holder to allow overriding behaviors."""
+
+
+class OAuth2ProviderTransactionTestCase(DjangoTransactionTestCase):
+    """Place holder to allow overriding behaviors."""
+
+
+if len(settings.DATABASES) > 1:
```

Comment:
I'm not a fan of import side effects. Can you encapsulated the db setup in  the  OAuth2ProviderTestCase and OAuth2ProviderTransactionTestCase or in a Mixin both can inherit?

Reply:
Let me explore that. Should be possible.

Reply:
The problem is some of the tests use pytest style fixtures and decorators instead of Django TestCases. This is why `test_database_names` and `database_for_oauth2_provider` are not defined within the TestCase classes. The pytest defined test functions need this information at import time.

The cleverness employed here is what allows the same set of tests to be single and multiple database tests. It is a trade off. I could get rid of the import time checks but I would need to duplicate the test cases which seems worse.
Definitely open to alternatives or other cleverness I could use.

Reply:
I think it complicates our testing infrastructure in a way that might not be obvious to new contributors and might not play nice with other decorators on our tests in the future.  I feel like there has to be a way to achieve the same functionality without having side effects.  Even most monkey patches come in a function flavor these days so they need to be explicitly applied and don't just happen on import. 

How difficult would it be to set self.databases and self.database_for_oauth2_provider in the __init__ of OAuth2ProviderTestCase and OAuth2ProviderTestCase, then use self.* where necessary rather than doing it as a side effect? 

Would making test_database_names a settings property be an option that solves the problem and allows it to be overrode on each test? 

or export a function that returns a value rather than conditionally declaring it on import?

Reply:
I can likely fix it for the TestCase based tests. It is the pytest based ones I am struggling with.